---
knit: neurotransmissionevolution::custom_knit
# output:
#   md_document:
#     variant: markdown_github
#     df_print: kable
#     dev: pdf
always_allow_html: yes
---

#### Creating tree graph\
The first step is to create a directed graph representing the NCBI Taxonomy tree.
```{r}
knitr::opts_chunk$set(
  warning     = FALSE,
  message     = FALSE,
  cache       = TRUE#,
  # cache.path  = "aaaa" 
)

library(here)
library(purrr)
library(tibble)

library(knitr)
library(kableExtra)

library(neurotransmissionevolution)

# download_if_missing <- function(filename, url) {
#   if (!file.exists(here("data-raw", "download", filename))) {
#     download.file(url, filename)
#   }
# }

# files <- tribble(
#   ~filename,           ~url,
#   "species.v11.0.txt", "https://stringdb-static.org/download/species.v11.0.txt",
#   # "download/taxonomy", "https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/new_taxdump/new_taxdump.tar.gz",
#   # "x",                 "y",
#   # "x",                 "y",
# )

files <- tribble(
  ~url,
  ~filename,
  
  "https://stringdb-static.org/download/species.v11.0.txt",
  "species.v11.0.txt",
  
  "https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/new_taxdump/new_taxdump.tar.gz",
  "new_taxdump.tar.gz",
  
  "http://rest.kegg.jp/link/pathway/hsa",
  "link_pathway_entrez.tsv",
  
  "https://string-db.org/mapping_files/entrez/human.entrez_2_string.2018.tsv.gz",
  "human.entrez_2_string.2018.tsv.gz",
  
  "https://string-db.org/mapping_files/STRING_display_names/human.name_2_string.tsv.gz",
  "human.name_2_string.tsv.gz",
  
  "https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz",
  "Homo_sapiens.gene_info.gz",
  
  "http://rest.genome.jp/link/ensembl/hsa",
  "link_ensembl_entrez.tsv",
  
  # "https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz",
  # "Homo_sapiens.gene_info.gz",
)

# files %>%
#   kable(format = "html") %>%
#   kable_styling(bootstrap_options = "striped")

files %>%
  kable("latex", booktabs = T) %>%
  kable_styling(latex_options = c("striped", "scale_down"))
```

* **test_table.tsv**
  * Just testing description
  1. **string_id**
    * **Description:** This represents the string protein identifier
    * **Type:** numeric
    * **Used:** yes
    * **Example:** 9606.ENSP000007157
  2. **length**
    * **Description:** This represents the protein size
    * **Type:** numeric
    * **Used:** no
    * **Example:** 462

First table

```{r}
spec <- tibble::tribble(
  ~col_name, ~col_type, ~used, ~ex, ~desc,
  "string_id", "numeric", "yes", "9606.ENSP000007157", paste("This represents the string protein identifier.",
                                                             "Like, really cool! Lorem ipsum dolar sit amet.",
                                                             "Niiiiii. Lero lero coronavirus coronaipsum."),
  "length", "numeric", "no", "462", "This represents the protein size",
)

# spec %>%
#   kable("latex", booktabs = T) %>%
#   add_header_above(c("test_table.tsv" = ncol(spec))) %>%
#   kable_styling(latex_options = c("striped", "scale_down")) %>%
#   column_spec(ncol(spec), width = "20em")
print_dataframe_specification(spec, "a", "b", "caption")
```

```{r}
# pwalk(files, download_if_missing)
plot(1,1)
```

```{r}
knitr::opts_chunk$get("cache.path")
# if (!file.exists("download/taxonomy")) {
#   tmp <- tempfile()
#   download.file("https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/new_taxdump/new_taxdump.tar.gz", tmp)
#   untar(tmp, exdir = "download/taxonomy")
#   unlink(tmp)
# }
```

