---
title: 'Gene identifiers and pathway annotation'
knit: neurotransmissionevolution::custom_knit
---
```{r message=FALSE, include=FALSE}
knitr::opts_chunk$set(
  warning     = FALSE,
  message     = FALSE,
  cache       = TRUE,
  cache.extra = file.mtime("download/link_pathway_entrez.tsv")
)
library(neurotransmissionevolution)
library(tidyverse)
# library(tidylog)
library(magrittr)
```

```{r download, include=FALSE}
download_if_missing("https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz")
download_if_missing("https://string-db.org/mapping_files/STRING_display_names/human.name_2_string.tsv.gz")
download_if_missing("https://string-db.org/mapping_files/entrez/human.entrez_2_string.2018.tsv.gz")
download_if_missing("http://rest.genome.jp/link/ensembl/hsa", "link_ensembl_entrez.tsv")
download_if_missing("http://rest.kegg.jp/link/pathway/hsa", "link_pathway_entrez.tsv")
```
### Neurotransmitter systems annotation
We start by querying the KEGG api for the pathways of interest. Resulting data is then pivoted to a wider format.
```{r results="asis", echo=FALSE}
link_pathway_entrez <- read_tsv(
  "download/link_pathway_entrez.tsv"
  ,col_names = c("entrez_id", "pathway_id")
  ,col_types = "cc"
)

link_pathway_entrez_spec <- tribble(
  ~col_name,    ~col_type,   ~used, ~ex,             ~desc,
  "entrez_id",  "character", "yes", "hsa:10411",     "NCBI Taxonomy identifier",
  "pathway_id", "character", "yes", "path:hsa04726", "KEGG pathway ID",
) %T>% print_dataframe_specification(
  "data-raw/download/link_pathway_entrez.tsv",
  "http://rest.kegg.jp/link/pathway/hsa",
  "All links between genes and pathways in KEGG.",
  "link_pathway_entrez"
)
```
```{r}
pathways <- tribble(
  ~pathway_id,      ~pathway_name, 
  "path:hsa04724",  "glutamatergic",
  "path:hsa04725",  "cholinergic",  
  "path:hsa04726",  "serotonergic", 
  "path:hsa04727",  "gabaergic",    
  "path:hsa04728",  "dopaminergic", 
  "path:hsa04721",  "vesicle"
)

# removing hsa prefix
link_pathway_entrez[["entrez_id"]] %<>% str_split_n(":", 2)

# filtering for pathways of interest and pivoting
gene_pathways <- inner_join(link_pathway_entrez, pathways) %>%
  mutate(n = 1) %>%
  pivot_wider(
    id_cols = entrez_id,
    names_from = pathway_name, 
    values_from = n,
    values_fn = list(n = length),
    values_fill = list(n = 0)
  ) %>%
  # filling 1's in all systems for synaptic vesicle genes
  # mutate_at(pathways[["pathway_name"]], ~ ifelse(vesicle == 1, 1L, .)) %>%
  mutate(system_count = rowSums(select(., -entrez_id, -vesicle)))

# exporting for package use
usethis::use_data(gene_pathways, overwrite = TRUE)
```
```{r echo=FALSE}
gene_pathways %>%
  tail %>%
  knitr::kable(booktabs = T) %>%
  kableExtra::kable_styling(latex_options = c("striped", "scale_down", "HOLD_position")) %>%
  kableExtra::add_header_above(c(`tail(gene_pathways)` = 8), monospace = T)
```


### Base ID lookup table
Now we start building a base ID lookup table containing entrez gene IDs, STRING ensembl protein IDs, ensembl gene IDs, STRING protein names and entrez gene names. Every piece of data in subsequent analyses will be progressively joined to it.
```{r results="asis", echo=FALSE}
##########################
## link_entrez_string
link_entrez_string <- read_tsv(
  "download/human.entrez_2_string.2018.tsv.gz",
  skip = 1,
  col_names = c("entrez_id", "string_id"),
  col_types = cols_only("-", "c", "c")
)
link_entrez_string_spec <- tribble(
  ~col_name,    ~col_type,   ~used, ~ex,                    ~desc,
  "taxid",      "numeric",   "no",  "9606",                 "NCBI Taxonomy ID",
  "entrez_id",  "numeric",   "yes", "7157",                 "entrez gene ID",
  "string_id",  "character", "yes", "9606.ENSP00000269305", "STRING ID",
) %T>% print_dataframe_specification(
  "data-raw/download/human.entrez_2_string.2018.tsv.gz",
  "https://string-db.org/mapping_files/entrez/human.entrez_2_string.2018.tsv.gz",
  "Conversion dictionary from entrez ID to STRING's ensembl protein ID.",
  "link_entrez_string"
)
##########################
## string_names
string_names <- read_tsv(
  "download/human.name_2_string.tsv.gz",
  skip = 1,
  col_names = c("string_name", "string_id"),
  col_types = cols_only("-", "c", "c")
)
string_names_spec <- tribble(
  ~col_name,     ~col_type,   ~used, ~ex,                    ~desc,
  "taxid",       "numeric",   "no",  "9606",                 "NCBI Taxonomy ID",
  "string_name", "character", "yes", "TP53",                 "protein name",
  "string_id",   "character", "yes", "9606.ENSP00000269305", "STRING ID",
) %T>% print_dataframe_specification(
  "data-raw/download/human.name_2_string.tsv.gz",
  "https://string-db.org/mapping_files/STRING_display_names/human.name_2_string.tsv.gz",
  "Conversion dictionary from STRING ID to protein name.",
  "string_names"
)
##########################
## entrez_names
entrez_names <- read_tsv(
  "download/Homo_sapiens.gene_info.gz",
  skip = 1,
  col_names = c("entrez_id", "entrez_name"),
  col_types = cols_only("-", "c", "c")
)
entrez_names_spec <- tribble(
  ~col_name,     ~col_type,   ~used, ~ex,    ~desc,
  "taxid",       "numeric",   "no",  "9606", "taxon ID",
  "entrez_id",   "character", "yes", "7157", "entrez gene ID",
  "entrez_name", "character", "yes", "TP53", "gene name",
  "...",         "...",       "no",  "...",  "(too many unrelated fields)",
) %T>% print_dataframe_specification(
  "data-raw/download/Homo_sapiens.gene_info.gz",
  "https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz",
  "Conversion dictionary from entrez ID to gene name.",
  "entrez_names"
)
##########################
## entrez_names
link_ensembl_entrez <- read_tsv(
  "download/link_ensembl_entrez.tsv",
  col_names = c("entrez_id", "ensembl_id"),
  col_types = "cc"
)
link_ensembl_entrez_spec <- tribble(
  ~col_name,     ~col_type,   ~used, ~ex,                       ~desc,
  "entrez_id",   "character", "yes", "hsa:7157",                "entrez gene ID",
  "ensembl_id",  "character", "yes", "ensembl:ENSG00000141510", "ensembl gene ID",
) %T>% print_dataframe_specification(
  "data-raw/download/link_ensembl_entrez.tsv",
  "http://rest.genome.jp/link/ensembl/hsa",
  "Conversion dictionary from entrez ID to ensembl gene (ENSG) ID.",
  "link_ensembl_entrez"
)
```
```{r}
# removing all kegg prefixes (e.g. "hsa:")
link_ensembl_entrez %<>% mutate_all(str_split_n, ":", 2)

# joining all data
gene_ids <- gene_pathways %>%
  select(entrez_id) %>%
  left_join(link_ensembl_entrez) %>%
  left_join(link_entrez_string) %>%
  left_join(string_names) %>%
  left_join(entrez_names)
```

Some STRING proteins couldn't be automatically resolved, so we perform it manually
```{r eval=FALSE}
gene_ids[!complete.cases(gene_ids),]
```
```{r results="asis", echo=FALSE}
gene_ids[!complete.cases(gene_ids),] %>%
  knitr::kable(booktabs = T) %>%
  kableExtra::kable_styling(position = "left", latex_options = c("striped", "HOLD_position"))
```

```{r}
complete_info <- tribble(
  ~entrez_id,    ~ensembl_id,        ~string_id,              ~string_name,     ~entrez_name,  
  "9296",       "ENSG00000128524",  "9606.ENSP00000417378",  "ATP6V1F",        "ATP6V1F",      
  "100137049",  "ENSG00000243708",  "9606.ENSP00000396045",  "PLA2G4B",        "PLA2G4B",      
  "85358",      "ENSG00000251322",  NA,                      NA,               "SHANK3",       
  "8681",       "ENSG00000168970",  "9606.ENSP00000371886",  "JMJD7-PLA2G4B",  "JMJD7-PLA2G4B",
  "1139",       "ENSG00000175344",  "9606.ENSP00000407546",  "CHRNA7",         "CHRNA7",       
  "107987478",  NA,                 NA,                      NA,               "LOC107987478", 
  "107987479",  NA,                 NA,                      NA,               "LOC107987479", 
  "1564",       "ENSG00000205702",  NA,                      NA,               "CYP2D7",       
  "801",        "ENSG00000198668",  "9606.ENSP00000349467",  "CALM1",          "CALM1",        
  "805",        "ENSG00000143933",  "9606.ENSP00000272298",  "CALM2",          "CALM2",        
  "808",        "ENSG00000160014",  "9606.ENSP00000291295",  "CALM3",          "CALM3"
)

# removing incomplete cases and adding updated ones
gene_ids %<>% na.omit %>% bind_rows(complete_info)

# removing taxid prefix from STRING IDs
gene_ids[["string_id"]] %<>% str_split_n("\\.", 2)

# exporting for package use
usethis::use_data(gene_ids, overwrite = TRUE)
```

