---
title: 'Network'
knit: neurotransmissionevolution::custom_knit
---
```{r include=FALSE}
knitr::opts_chunk$set(
   warning     = FALSE
  ,message     = FALSE
  ,cache       = 1
  ,autodep     = T
) 
library(RCurl)
library(tidyverse)
library(tidylog)
library(magrittr)
library(neurotransmissionevolution)

load("../data/gene_ids.rda")
```

### Retrieving network data
Querying the API endpoint for the STRING IDs we previously resolved:
```{r}
identifiers <- gene_ids %>% pull(string_id) %>% na.omit %>% paste0(collapse="%0d")

if (!file.exists("download/string_ids.tsv")) {
    postForm(
      "http://string-db.org/api/tsv/get_string_ids"
      ,identifiers = identifiers
      ,echo_query  = "1"
      ,species     = "9606"
    ) %>%
    write("download/string_ids.tsv")
}
```
```{r results="asis", echo=FALSE}
api_ids_spec <- tibble::tribble(
      ~col_name,   ~col_type, ~used,                                      ~ex,                   ~desc,
    "queryItem", "character", "yes",                        "ENSP00000258400",          "queried term",
   "queryIndex",   "numeric", "yes",                                    "266", "index of queried term",
     "stringId", "character", "yes",                   "9606.ENSP00000258400",             "STRING ID",
  "ncbiTaxonId",   "numeric", "yes",                                   "9606",      "NCBI Taxonomy ID",
    "taxonName", "character", "yes",                           "Homo sapiens",          "species name",
"preferredName", "character", "yes",                                  "HTR2B",   "common protein name",
   "annotation", "character", "yes", "5-hydroxytryptamine receptor 2B; [...]",    "protein annotation"
) %T>% print_dataframe_specification(
  "data-raw/download/string_ids.tsv",
  "http://string-db.org/api/tsv/get_string_ids",
  "STRING interaction network with channel specific scores.",
  "string_ids"
)
```

Making sure that the API succesfully resolves queried protein IDs.
```{r}
api_ids <- read_tsv("download/string_ids.tsv", comment = "", quote = "")

# Removing taxid prefix
api_ids %<>% mutate(stringId = str_split_n(stringId, "\\.", 2))

# Removing inexact matches (queried id is different from resolved id)
api_ids %<>% group_by(queryItem) %>% filter(queryItem == stringId)

# Setequal must return true if ids matched exatcly
setequal(
  gene_ids %>% pull(string_id) %>% na.omit,
  api_ids  %>% pull(stringId)
) %>% stopifnot
```

Once IDs are correctly resolved, we can query the network API endpoint to obtain the protein-protein interaction edgelist.
```{r}
# It is important to query this endpoint with the species prefix ("9606.")
identifiers <- api_ids %>% pull(stringId) %>% na.omit %>% { paste0("9606.", ., collapse="%0d") }

if (!file.exists("download/string_edgelist.tsv")) {
    postForm(
      "http://string-db.org/api/tsv/network"
      ,identifiers = identifiers
      ,species     = "9606"
    ) %>%
    write("download/string_edgelist.tsv")
}
```
```{r results="asis", echo=FALSE}
string_edgelist_spec <- tibble::tribble(
          ~col_name,   ~col_type, ~used,               ~ex,                             ~desc,
       "stringId_A", "character", "yes", "ENSP00000215659",           "STRING ID (protein A)",
       "stringId_B", "character", "yes", "ENSP00000211287",           "STRING ID (protein B)",
  "preferredName_A", "character", "yes",          "MAPK12", "common protein name (protein A)",
  "preferredName_B", "character", "yes",          "MAPK13", "common protein name (protein B)",
      "ncbiTaxonId",   "numeric", "yes",            "9606",                "NCBI Taxonomy ID",
            "score",   "numeric", "yes",           "0.948",                  "combined score",
           "nscore",   "numeric", "yes",               "0",         "gene neighborhood score",
           "fscore",   "numeric", "yes",               "0",               "gene fusion score",
           "pscore",   "numeric", "yes",        "0.014223",      "phylogenetic profile score",
           "ascore",   "numeric", "yes",               "0",              "coexpression score",
           "escore",   "numeric", "yes",           "0.485",              "experimental score",
           "dscore",   "numeric", "yes",             "0.9",                  "database score",
           "tscore",   "numeric", "yes",         "0.02772",                "textmining score"
) %T>% print_dataframe_specification(
  "data-raw/download/string_edgelist.tsv",
  "http://string-db.org/api/tsv/network",
  "STRING interaction network with channel specific scores.",
  "string_edgelist"
)
```

### Recomputing scores
From [string-db.org](https://string-db.org/cgi/info.pl?footer_active_subpage=scores):

> "In STRING, each protein-protein interaction is annotated with one or more 'scores'.
Importantly, these scores do not indicate the strength or the specificity of the interaction. Instead, they are indicators of confidence, i.e. how likely STRING judges an interaction to be true, given the available evidence. All scores rank from 0 to 1, with 1 being the highest possible confidence."

For the sake of this project, we will only use experimental and database scores with a combined value >= 0.7, a high confidence threshold according to the STRING database. The combined score is given by the following expression, as stated in [von Mering C et al, 2005](https://doi.org/10.1093/nar/gki005):
$$S\ =\ 1\ {-}\ {{\prod}_{i}}\left(1\ {-}\ S_{i}\right)$$
```{r results="hide", echo=TRUE}
string_edgelist <- read_tsv("download/string_edgelist.tsv")

string_edgelist %<>%
  mutate(cs = combine_scores(., c("e","d"))) %>%
  filter(cs >= 0.7) %>%
  select(stringId_A, stringId_B)

# How many edgelist proteins are absent in gene_ids? (should return 0)
setdiff(
  string_edgelist %$% c(stringId_A, stringId_B),
  gene_ids %>% pull(string_id)
) %>% { stopifnot(. == 0) }

# Exporting for package use
usethis::use_data(string_edgelist, overwrite = TRUE)
```
