---
title: "NCBI Taxonomy tree"
knit: neurotransmissionevolution::custom_knit
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(
  warning     = FALSE,
  message     = FALSE,
  cache       = TRUE,
  cache.extra = file.mtime("download/taxdump.tar.gz")
) 

library(tidyverse)
library(igraph)
# library(tidylog)
library(tibble)
library(magrittr)

library(neurotransmissionevolution)
```

#### Downloading data
```{r message=FALSE}
download_if_missing("http://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz")
download_if_missing("stringdb-static.org/download/species.v11.0.txt")

untar("download/taxdump.tar.gz", exdir = "download/taxdump")
```

#### Loading data
```{r results='asis', echo=FALSE}
###################
## string_species
###################
string_species_spec <- tribble(
  ~col_name,            ~col_type,   ~used, ~ex,            ~desc,
  "taxid",           "character", "yes", "9606",         "NCBI Taxonomy identifier",
  "string_type",        "character", "no",  "core",         "if the genome of this species is core or periphery",
  "string_name",        "character", "yes", "Homo sapiens", "STRING species name",
  "ncbi_official_name", "character", "no",  "Homo sapiens", "NCBI Taxonomy species name"
) %T>% print_dataframe_specification(
  "data-raw/download/species.v11.0.txt",
  "stringdb-static.org/download/species.v11.0.txt",
  "Lists all organisms in STRING v11."
)
###################
## ncbi_merged_ids
###################
ncbi_merged_ids_spec <- tribble(
  ~col_name,   ~col_type,   ~used,  ~ex,    ~desc,
  "taxid",     "character", "yes",  "9606", "id of node that has been merged",
  "new_taxid", "character", "yes",  "core", "id of node that is the result of merging",
) %T>% print_dataframe_specification(
  "data-raw/download/taxdump/merged.dmp",
  "ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz",
  "Links outdated taxon IDs to corresponding new ones."
)
###################
## ncbi_edgelist
###################
ncbi_edgelist_spec <- tribble(
  ~col_name,       ~col_type,  ~used,  ~ex,            ~desc,
  "taxid",        "character", "yes",  "2",            "node id in NCBI taxonomy database",
  "parent_taxid", "character", "yes",  "131567",       "parent node id in NCBI taxonomy database",
  "rank",         "character", "no",   "superkingdom", "rank of this node",
  "...",          "",          "no",   "",             "(too many unrelated fields)",
) %T>% print_dataframe_specification(
  "data-raw/download/taxdump/nodes.dmp",
  "ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz",
  "Represents taxonomy nodes."
)
###################
## ncbi_taxon_names
###################
ncbi_taxon_names_spec <- tribble(
  ~col_name,     ~col_type,  ~used,  ~ex,                  ~desc,
  "taxid",       "character", "yes",  "2",                 "the id of node associated with this name",
  "name",        "character", "yes",  "Monera",            "name itself",
  "unique_name", "character", "no",   "Monera <bacteria>", "the unique variant of this name if name not unique",
  "name_class",  "character", "yes",  "scientific name",   "type of name",
) %T>% print_dataframe_specification(
  "data-raw/download/taxdump/names.dmp",
  "ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz",
  "Links taxon IDs to actual species names."
)
```

```{r message=FALSE, warning=FALSE}
string_species <- read_tsv(
  "download/species.v11.0.txt",
  skip = 1,
  col_names = c(
    "taxid",
    "string_type",
    "string_name",
    "ncbi_official_name"
  ),
  col_types = cols_only(
    taxid = "c",
    string_name = "c"
  )
)

# these .dmp files are very tricky to read
# the following read_delims are very hacky
ncbi_merged_ids <- read_delim(
  "download/taxdump/merged.dmp",
  delim = "|",
  trim_ws = TRUE,
  col_names = c("taxid","new_taxid"),
  col_types = "cc"
)

ncbi_edgelist <- read_delim(
  "download/taxdump/nodes.dmp",
  skip = 1,
  delim = "|",
  trim_ws = TRUE,
  col_names = c("n1","n2"),
  col_types = "cc"
)

ncbi_taxon_names <- read_delim(
  "download/taxdump/names.dmp",
  delim = "|",
  trim_ws = TRUE,
  col_names = c("name","ncbi_name","type"),
  col_types = "cc-c"
)
```

#### Updating STRING taxon IDs\
Some organisms taxon IDs are outdated in STRING. We must update them to work with the most recent NCBI Taxonomy data.
```{r}
string_species %<>%
  left_join(ncbi_merged_ids) %>%
  mutate(new_taxid = coalesce(new_taxid, taxid))
```

#### Creating tree graph\
The first step is to create a directed graph representing the NCBI Taxonomy tree.
```{r}
# leaving only "scientific name" rows
ncbi_taxon_names %<>%
  filter(type == "scientific name") %>%
  select(name, ncbi_name)

# finding Eukaryota taxid
eukaryota_taxon_id <- subset(ncbi_taxon_names, ncbi_name == "Eukaryota", "name", drop = TRUE)

# creating graph
g <- graph_from_data_frame(ncbi_edgelist[,2:1], directed = TRUE, vertices = ncbi_taxon_names)

# easing memory
rm(ncbi_edgelist, ncbi_merged_ids)
```

#### Traversing the graph\
The second step is to traverse the graph from the Eukaryota root node to STRING species nodes. This automatically drops all non-eukaryotes and results in a species tree representing only STRING eukaryotes (476).
```{r}
eukaryote_root <- V(g)[eukaryota_taxon_id]
eukaryote_leaves <- V(g)[string_species[["new_taxid"]]]

# not_found <- subset(string_species, !new_taxid %in% ncbi_taxon_names$name)

eukaryote_paths <- shortest_paths(g, from = eukaryote_root, to = eukaryote_leaves, mode = "out")$vpath

eukaryote_vertices <- eukaryote_paths %>% unlist %>% unique

eukaryote_tree <- induced_subgraph(g, eukaryote_vertices, impl = "create_from_scratch")
```

#### Saving\
Saving `ncbi_tree` and `string_eukaryotes` for package use. These data files are documented by the package.
We also create a plain text file `476_ncbi_eukaryotes.txt` containing the updated names of all 476 STRING eukaryotes. This file will be queried against the TimeTree website.
```{r}
ncbi_tree <- treeio::as.phylo(eukaryote_tree)

# plot(ncbi_tree %>% ape::ladderize(), type="cladogram")

string_eukaryotes <- string_species %>%
  filter(new_taxid %in% ncbi_tree$tip.label) %>%
  inner_join(ncbi_taxon_names, by = c("new_taxid" = "name"))

write(string_eukaryotes[["ncbi_name"]],"476_ncbi_eukaryotes.txt")

usethis::use_data(ncbi_tree, overwrite = TRUE)
usethis::use_data(string_eukaryotes, overwrite = TRUE)
```
