\hypertarget{retrieving-network-data}{%
\subsubsection{Retrieving network data}\label{retrieving-network-data}}

Querying the API endpoint for the STRING IDs we previously resolved:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{identifiers <-}\StringTok{ }\NormalTok{gene_ids }\OperatorTok{%>%}\StringTok{ }\KeywordTok{pull}\NormalTok{(string_id) }\OperatorTok{%>%}\StringTok{ }\NormalTok{na.omit }\OperatorTok{%>%}\StringTok{ }\KeywordTok{paste0}\NormalTok{(}\DataTypeTok{collapse=}\StringTok{"%0d"}\NormalTok{)}

\ControlFlowTok{if}\NormalTok{ (}\OperatorTok{!}\KeywordTok{file.exists}\NormalTok{(}\StringTok{"download/string_ids.tsv"}\NormalTok{)) \{}
    \KeywordTok{postForm}\NormalTok{(}
      \StringTok{"http://string-db.org/api/tsv/get_string_ids"}
\NormalTok{      ,}\DataTypeTok{identifiers =}\NormalTok{ identifiers}
\NormalTok{      ,}\DataTypeTok{echo_query  =} \StringTok{"1"}
\NormalTok{      ,}\DataTypeTok{species     =} \StringTok{"9606"}
\NormalTok{    ) }\OperatorTok{%>%}
\StringTok{    }\KeywordTok{write}\NormalTok{(}\StringTok{"download/string_ids.tsv"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{table}[H]

\caption{\label{tab:string_ids}STRING interaction network with channel specific scores.}
\begin{tabular}[t]{rlllll}
\toprule
\multicolumn{6}{c}{\bgroup\fontsize{12}{14}\selectfont \cellcolor[HTML]{EEEEEE}{\ttfamily{\textbf{string\_ids}}}\egroup{}} \\
\cmidrule(l{3pt}r{3pt}){1-6}
\# & Col. name & Col. type & Used? & Example & Description\\
\midrule
\rowcolor{gray!6}  1 & queryItem & character & yes & ENSP00000258400 & queried term\\
2 & queryIndex & numeric & yes & 266 & index of queried term\\
\rowcolor{gray!6}  3 & stringId & character & yes & 9606.ENSP00000258400 & STRING ID\\
4 & ncbiTaxonId & numeric & yes & 9606 & NCBI Taxonomy ID\\
\rowcolor{gray!6}  5 & taxonName & character & yes & Homo sapiens & species name\\
\addlinespace
6 & preferredName & character & yes & HTR2B & common protein name\\
\rowcolor{gray!6}  7 & annotation & character & yes & 5-hydroxytryptamine receptor 2B; [...] & protein annotation\\
\bottomrule
\multicolumn{6}{l}{\textbf{Location: } data-raw/download/string\_ids.tsv}\\
\multicolumn{6}{l}{\textbf{Source: } http://string-db.org/api/tsv/get\_string\_ids}\\
\end{tabular}
\end{table}

Making sure that the API succesfully resolves queried protein IDs.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{api_ids <-}\StringTok{ }\KeywordTok{read_tsv}\NormalTok{(}\StringTok{"download/string_ids.tsv"}\NormalTok{, }\DataTypeTok{comment =} \StringTok{""}\NormalTok{, }\DataTypeTok{quote =} \StringTok{""}\NormalTok{)}

\CommentTok{# Removing taxid prefix}
\NormalTok{api_ids }\OperatorTok{%<>%}\StringTok{ }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{stringId =} \KeywordTok{str_split_n}\NormalTok{(stringId, }\StringTok{"}\CharTok{\textbackslash{}\textbackslash{}}\StringTok{."}\NormalTok{, }\DecValTok{2}\NormalTok{))}

\CommentTok{# Removing inexact matches (queried id is different from resolved id)}
\NormalTok{api_ids }\OperatorTok{%<>%}\StringTok{ }\KeywordTok{group_by}\NormalTok{(queryItem) }\OperatorTok{%>%}\StringTok{ }\KeywordTok{filter}\NormalTok{(queryItem }\OperatorTok{==}\StringTok{ }\NormalTok{stringId)}

\CommentTok{# Setequal must return true if ids matched exatcly}
\KeywordTok{setequal}\NormalTok{(}
\NormalTok{  gene_ids }\OperatorTok{%>%}\StringTok{ }\KeywordTok{pull}\NormalTok{(string_id) }\OperatorTok{%>%}\StringTok{ }\NormalTok{na.omit,}
\NormalTok{  api_ids  }\OperatorTok{%>%}\StringTok{ }\KeywordTok{pull}\NormalTok{(stringId)}
\NormalTok{) }\OperatorTok{%>%}\StringTok{ }\NormalTok{stopifnot}
\end{Highlighting}
\end{Shaded}

Once IDs are correctly resolved, we can query the network API endpoint
to obtain the protein-protein interaction edgelist.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# It is important to query this endpoint with the species prefix ("9606.")}
\NormalTok{identifiers <-}\StringTok{ }\NormalTok{api_ids }\OperatorTok{%>%}\StringTok{ }\KeywordTok{pull}\NormalTok{(stringId) }\OperatorTok{%>%}\StringTok{ }\NormalTok{na.omit }\OperatorTok{%>%}\StringTok{ }\NormalTok{\{ }\KeywordTok{paste0}\NormalTok{(}\StringTok{"9606."}\NormalTok{, ., }\DataTypeTok{collapse=}\StringTok{"%0d"}\NormalTok{) \}}

\ControlFlowTok{if}\NormalTok{ (}\OperatorTok{!}\KeywordTok{file.exists}\NormalTok{(}\StringTok{"download/string_edgelist.tsv"}\NormalTok{)) \{}
    \KeywordTok{postForm}\NormalTok{(}
      \StringTok{"http://string-db.org/api/tsv/network"}
\NormalTok{      ,}\DataTypeTok{identifiers =}\NormalTok{ identifiers}
\NormalTok{      ,}\DataTypeTok{species     =} \StringTok{"9606"}
\NormalTok{    ) }\OperatorTok{%>%}
\StringTok{    }\KeywordTok{write}\NormalTok{(}\StringTok{"download/string_edgelist.tsv"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{table}[H]

\caption{\label{tab:string_edgelist}STRING interaction network with channel specific scores.}
\begin{tabular}[t]{rlllll}
\toprule
\multicolumn{6}{c}{\bgroup\fontsize{12}{14}\selectfont \cellcolor[HTML]{EEEEEE}{\ttfamily{\textbf{string\_edgelist}}}\egroup{}} \\
\cmidrule(l{3pt}r{3pt}){1-6}
\# & Col. name & Col. type & Used? & Example & Description\\
\midrule
\rowcolor{gray!6}  1 & stringId\_A & character & yes & ENSP00000215659 & STRING ID (protein A)\\
2 & stringId\_B & character & yes & ENSP00000211287 & STRING ID (protein B)\\
\rowcolor{gray!6}  3 & preferredName\_A & character & yes & MAPK12 & common protein name (protein A)\\
4 & preferredName\_B & character & yes & MAPK13 & common protein name (protein B)\\
\rowcolor{gray!6}  5 & ncbiTaxonId & numeric & yes & 9606 & NCBI Taxonomy ID\\
\addlinespace
6 & score & numeric & yes & 0.948 & combined score\\
\rowcolor{gray!6}  7 & nscore & numeric & yes & 0 & gene neighborhood score\\
8 & fscore & numeric & yes & 0 & gene fusion score\\
\rowcolor{gray!6}  9 & pscore & numeric & yes & 0.014223 & phylogenetic profile score\\
10 & ascore & numeric & yes & 0 & coexpression score\\
\addlinespace
\rowcolor{gray!6}  11 & escore & numeric & yes & 0.485 & experimental score\\
12 & dscore & numeric & yes & 0.9 & database score\\
\rowcolor{gray!6}  13 & tscore & numeric & yes & 0.02772 & textmining score\\
\bottomrule
\multicolumn{6}{l}{\textbf{Location: } data-raw/download/string\_edgelist.tsv}\\
\multicolumn{6}{l}{\textbf{Source: } http://string-db.org/api/tsv/network}\\
\end{tabular}
\end{table}

\hypertarget{recomputing-scores}{%
\subsubsection{Recomputing scores}\label{recomputing-scores}}

From
\href{https://string-db.org/cgi/info.pl?footer_active_subpage=scores}{string-db.org}:

\begin{quote}
``In STRING, each protein-protein interaction is annotated with one or
more `scores'. Importantly, these scores do not indicate the strength or
the specificity of the interaction. Instead, they are indicators of
confidence, i.e.~how likely STRING judges an interaction to be true,
given the available evidence. All scores rank from 0 to 1, with 1 being
the highest possible confidence.''
\end{quote}

For the sake of this project, we will only use experimental and database
scores with a combined value \textgreater= 0.7, a high confidence
threshold according to the STRING database. The combined score is given
by the following expression, as stated in
\href{https://doi.org/10.1093/nar/gki005}{von Mering C et al, 2005}:
\[S\ =\ 1\ {-}\ {{\prod}_{i}}\left(1\ {-}\ S_{i}\right)\]

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{string_edgelist <-}\StringTok{ }\KeywordTok{read_tsv}\NormalTok{(}\StringTok{"download/string_edgelist.tsv"}\NormalTok{)}

\NormalTok{string_edgelist }\OperatorTok{%<>%}
\StringTok{  }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{cs =} \KeywordTok{combine_scores}\NormalTok{(., }\KeywordTok{c}\NormalTok{(}\StringTok{"e"}\NormalTok{,}\StringTok{"d"}\NormalTok{))) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{filter}\NormalTok{(cs }\OperatorTok{>=}\StringTok{ }\FloatTok{0.7}\NormalTok{) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{select}\NormalTok{(stringId_A, stringId_B)}

\CommentTok{# How many edgelist proteins are absent in gene_ids? (should return 0)}
\KeywordTok{setdiff}\NormalTok{(}
\NormalTok{  string_edgelist }\OperatorTok{%$%}\StringTok{ }\KeywordTok{c}\NormalTok{(stringId_A, stringId_B),}
\NormalTok{  gene_ids }\OperatorTok{%>%}\StringTok{ }\KeywordTok{pull}\NormalTok{(string_id)}
\NormalTok{) }\OperatorTok{%>%}\StringTok{ }\NormalTok{\{ }\KeywordTok{stopifnot}\NormalTok{(. }\OperatorTok{==}\StringTok{ }\DecValTok{0}\NormalTok{) \}}

\CommentTok{# Exporting for package use}
\NormalTok{usethis}\OperatorTok{::}\KeywordTok{use_data}\NormalTok{(string_edgelist, }\DataTypeTok{overwrite =} \OtherTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

