\hypertarget{neurotransmitter-systems-annotation}{%
\subsubsection{Neurotransmitter systems
annotation}\label{neurotransmitter-systems-annotation}}

We start by querying the KEGG api for the pathways of interest. Results
are pivoted to a wider format.

\begin{table}[H]

\caption{\label{tab:link_pathway_entrez}All links between genes and pathways in KEGG.}
\begin{tabular}[t]{rlllll}
\toprule
\multicolumn{6}{c}{\bgroup\fontsize{12}{14}\selectfont \cellcolor[HTML]{EEEEEE}{\ttfamily{\textbf{link\_pathway\_entrez}}}\egroup{}} \\
\cmidrule(l{3pt}r{3pt}){1-6}
\# & Col. name & Col. type & Used? & Example & Description\\
\midrule
\rowcolor{gray!6}  1 & entrez\_id & character & yes & hsa:10411 & NCBI Taxonomy identifier\\
2 & pathway\_id & character & yes & path:hsa04726 & KEGG pathway ID\\
\bottomrule
\multicolumn{6}{l}{\textbf{Location: } data-raw/download/link\_pathway\_entrez.tsv}\\
\multicolumn{6}{l}{\textbf{Source: } http://rest.kegg.jp/link/pathway/hsa}\\
\end{tabular}
\end{table}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pathways <-}\StringTok{ }\KeywordTok{tribble}\NormalTok{(}
  \OperatorTok{~}\NormalTok{pathway_id,      }\OperatorTok{~}\NormalTok{pathway_name, }
  \StringTok{"path:hsa04724"}\NormalTok{,  }\StringTok{"glutamatergic"}\NormalTok{,}
  \StringTok{"path:hsa04725"}\NormalTok{,  }\StringTok{"cholinergic"}\NormalTok{,  }
  \StringTok{"path:hsa04726"}\NormalTok{,  }\StringTok{"serotonergic"}\NormalTok{, }
  \StringTok{"path:hsa04727"}\NormalTok{,  }\StringTok{"gabaergic"}\NormalTok{,    }
  \StringTok{"path:hsa04728"}\NormalTok{,  }\StringTok{"dopaminergic"}
\NormalTok{)}

\CommentTok{# Removing hsa prefix}
\NormalTok{link_pathway_entrez[[}\StringTok{"entrez_id"}\NormalTok{]] }\OperatorTok{%<>%}\StringTok{ }\KeywordTok{str_split_n}\NormalTok{(}\StringTok{":"}\NormalTok{, }\DecValTok{2}\NormalTok{)}

\CommentTok{# Filtering for pathways of interest and pivoting}
\NormalTok{gene_pathways <-}\StringTok{ }\KeywordTok{inner_join}\NormalTok{(link_pathway_entrez, pathways) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{n =} \DecValTok{1}\NormalTok{) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{pivot_wider}\NormalTok{(}
    \DataTypeTok{id_cols     =}\NormalTok{ entrez_id,}
    \DataTypeTok{names_from  =}\NormalTok{ pathway_name, }
    \DataTypeTok{values_from =}\NormalTok{ n,}
    \DataTypeTok{values_fn   =} \KeywordTok{list}\NormalTok{(}\DataTypeTok{n =}\NormalTok{ length),}
    \DataTypeTok{values_fill =} \KeywordTok{list}\NormalTok{(}\DataTypeTok{n =} \DecValTok{0}\NormalTok{)}
\NormalTok{  ) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{system_count =} \KeywordTok{rowSums}\NormalTok{(}\KeywordTok{select}\NormalTok{(., }\OperatorTok{-}\NormalTok{entrez_id)))}

\CommentTok{# Exporting for package use}
\NormalTok{usethis}\OperatorTok{::}\KeywordTok{use_data}\NormalTok{(gene_pathways, }\DataTypeTok{overwrite =} \OtherTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}[H]
\centering
\resizebox{\linewidth}{!}{
\begin{tabular}{lrrrrrr}
\toprule
\multicolumn{7}{c}{\ttfamily{tail(gene\_pathways)}} \\
\cmidrule(l{3pt}r{3pt}){1-7}
entrez\_id & glutamatergic & cholinergic & serotonergic & gabaergic & dopaminergic & system\_count\\
\midrule
\rowcolor{gray!6}  805 & 0 & 0 & 0 & 0 & 1 & 1\\
808 & 0 & 0 & 0 & 0 & 1 & 1\\
\rowcolor{gray!6}  810 & 0 & 0 & 0 & 0 & 1 & 1\\
84152 & 0 & 0 & 0 & 0 & 1 & 1\\
\rowcolor{gray!6}  91860 & 0 & 0 & 0 & 0 & 1 & 1\\
\addlinespace
9575 & 0 & 0 & 0 & 0 & 1 & 1\\
\bottomrule
\end{tabular}}
\end{table}

\hypertarget{base-id-lookup-table}{%
\subsubsection{Base ID lookup table}\label{base-id-lookup-table}}

Now we start building a base ID lookup table containing entrez gene IDs,
STRING ensembl protein IDs, ensembl gene IDs, STRING protein names and
entrez gene names. Every piece of data in subsequent analyses will be
progressively joined to it.

\begin{table}[H]

\caption{\label{tab:link_entrez_string}Conversion dictionary from entrez ID to STRING's ensembl protein ID.}
\begin{tabular}[t]{rlllll}
\toprule
\multicolumn{6}{c}{\bgroup\fontsize{12}{14}\selectfont \cellcolor[HTML]{EEEEEE}{\ttfamily{\textbf{link\_entrez\_string}}}\egroup{}} \\
\cmidrule(l{3pt}r{3pt}){1-6}
\# & Col. name & Col. type & Used? & Example & Description\\
\midrule
\rowcolor{gray!6}  1 & taxid & numeric & no & 9606 & NCBI Taxonomy ID\\
2 & entrez\_id & numeric & yes & 7157 & entrez gene ID\\
\rowcolor{gray!6}  3 & string\_id & character & yes & 9606.ENSP00000269305 & STRING ID\\
\bottomrule
\multicolumn{6}{l}{\textbf{Location: } data-raw/download/human.entrez\_2\_string.2018.tsv.gz}\\
\multicolumn{6}{l}{\textbf{Source: } https://string-db.org/mapping\_files/entrez/human.entrez\_2\_string.2018.tsv.gz}\\
\end{tabular}
\end{table}
\begin{table}[H]

\caption{\label{tab:string_names}Conversion dictionary from STRING ID to protein name.}
\begin{tabular}[t]{rlllll}
\toprule
\multicolumn{6}{c}{\bgroup\fontsize{12}{14}\selectfont \cellcolor[HTML]{EEEEEE}{\ttfamily{\textbf{string\_names}}}\egroup{}} \\
\cmidrule(l{3pt}r{3pt}){1-6}
\# & Col. name & Col. type & Used? & Example & Description\\
\midrule
\rowcolor{gray!6}  1 & taxid & numeric & no & 9606 & NCBI Taxonomy ID\\
2 & string\_name & character & yes & TP53 & protein name\\
\rowcolor{gray!6}  3 & string\_id & character & yes & 9606.ENSP00000269305 & STRING ID\\
\bottomrule
\multicolumn{6}{l}{\textbf{Location: } data-raw/download/human.name\_2\_string.tsv.gz}\\
\multicolumn{6}{l}{\textbf{Source: } https://string-db.org/mapping\_files/STRING\_display\_names/human.name\_2\_string.tsv.gz}\\
\end{tabular}
\end{table}
\begin{table}[H]

\caption{\label{tab:entrez_names}Conversion dictionary from entrez ID to gene name.}
\begin{tabular}[t]{rlllll}
\toprule
\multicolumn{6}{c}{\bgroup\fontsize{12}{14}\selectfont \cellcolor[HTML]{EEEEEE}{\ttfamily{\textbf{entrez\_names}}}\egroup{}} \\
\cmidrule(l{3pt}r{3pt}){1-6}
\# & Col. name & Col. type & Used? & Example & Description\\
\midrule
\rowcolor{gray!6}  1 & taxid & numeric & no & 9606 & taxon ID\\
2 & entrez\_id & character & yes & 7157 & entrez gene ID\\
\rowcolor{gray!6}  3 & entrez\_name & character & yes & TP53 & gene name\\
4 & ... & ... & no & ... & (too many unrelated fields)\\
\bottomrule
\multicolumn{6}{l}{\textbf{Location: } data-raw/download/Homo\_sapiens.gene\_info.gz}\\
\multicolumn{6}{l}{\textbf{Source: } https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE\_INFO/Mammalia/Homo\_sapiens.gene\_info.gz}\\
\end{tabular}
\end{table}
\begin{table}[H]

\caption{\label{tab:link_ensembl_entrez}Conversion dictionary from entrez ID to ensembl gene (ENSG) ID.}
\begin{tabular}[t]{rlllll}
\toprule
\multicolumn{6}{c}{\bgroup\fontsize{12}{14}\selectfont \cellcolor[HTML]{EEEEEE}{\ttfamily{\textbf{link\_ensembl\_entrez}}}\egroup{}} \\
\cmidrule(l{3pt}r{3pt}){1-6}
\# & Col. name & Col. type & Used? & Example & Description\\
\midrule
\rowcolor{gray!6}  1 & entrez\_id & character & yes & hsa:7157 & entrez gene ID\\
2 & ensembl\_id & character & yes & ensembl:ENSG00000141510 & ensembl gene ID\\
\bottomrule
\multicolumn{6}{l}{\textbf{Location: } data-raw/download/link\_ensembl\_entrez.tsv}\\
\multicolumn{6}{l}{\textbf{Source: } http://rest.genome.jp/link/ensembl/hsa}\\
\end{tabular}
\end{table}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# Removing all kegg prefixes (e.g. "hsa:")}
\NormalTok{link_ensembl_entrez }\OperatorTok{%<>%}\StringTok{ }\KeywordTok{mutate_all}\NormalTok{(str_split_n, }\StringTok{":"}\NormalTok{, }\DecValTok{2}\NormalTok{)}

\CommentTok{# Joining all data}
\NormalTok{gene_ids <-}\StringTok{ }\NormalTok{gene_pathways }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{select}\NormalTok{(entrez_id) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{left_join}\NormalTok{(link_ensembl_entrez) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{left_join}\NormalTok{(link_entrez_string) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{left_join}\NormalTok{(string_names) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{left_join}\NormalTok{(entrez_names)}
\end{Highlighting}
\end{Shaded}

Some STRING proteins couldn't be automatically resolved, so we resolve
them manually

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gene_ids[}\OperatorTok{!}\KeywordTok{complete.cases}\NormalTok{(gene_ids),]}
\end{Highlighting}
\end{Shaded}

\begin{tabular}{lllll}
\toprule
entrez\_id & ensembl\_id & string\_id & string\_name & entrez\_name\\
\midrule
\rowcolor{gray!6}  100137049 & ENSG00000243708 & NA & NA & PLA2G4B\\
85358 & ENSG00000251322 & NA & NA & SHANK3\\
\rowcolor{gray!6}  8681 & ENSG00000168970 & NA & NA & JMJD7-PLA2G4B\\
1139 & ENSG00000175344 & NA & NA & CHRNA7\\
\rowcolor{gray!6}  107987478 & NA & NA & NA & LOC107987478\\
\addlinespace
107987479 & NA & NA & NA & LOC107987479\\
\rowcolor{gray!6}  1564 & ENSG00000205702 & NA & NA & CYP2D7\\
801 & ENSG00000198668 & NA & NA & CALM1\\
\rowcolor{gray!6}  805 & ENSG00000143933 & NA & NA & CALM2\\
808 & ENSG00000160014 & NA & NA & CALM3\\
\bottomrule
\end{tabular}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{complete_info <-}\StringTok{ }\KeywordTok{tribble}\NormalTok{(}
\CommentTok{##########################################################################################}
 \OperatorTok{~}\NormalTok{entrez_id,       }\OperatorTok{~}\NormalTok{ensembl_id,             }\OperatorTok{~}\NormalTok{string_id,    }\OperatorTok{~}\NormalTok{string_name,    }\OperatorTok{~}\NormalTok{entrez_name,}\CommentTok{#}
\StringTok{"100137049"}\NormalTok{, }\StringTok{"ENSG00000243708"}\NormalTok{, }\StringTok{"9606.ENSP00000396045"}\NormalTok{,       }\StringTok{"PLA2G4B"}\NormalTok{,       }\StringTok{"PLA2G4B"}\NormalTok{,}\CommentTok{#}
    \StringTok{"85358"}\NormalTok{, }\StringTok{"ENSG00000251322"}\NormalTok{,                     }\OtherTok{NA}\NormalTok{,              }\OtherTok{NA}\NormalTok{,        }\StringTok{"SHANK3"}\NormalTok{,}\CommentTok{#}
     \StringTok{"8681"}\NormalTok{, }\StringTok{"ENSG00000168970"}\NormalTok{, }\StringTok{"9606.ENSP00000371886"}\NormalTok{, }\StringTok{"JMJD7-PLA2G4B"}\NormalTok{, }\StringTok{"JMJD7-PLA2G4B"}\NormalTok{,}\CommentTok{#}
     \StringTok{"1139"}\NormalTok{, }\StringTok{"ENSG00000175344"}\NormalTok{, }\StringTok{"9606.ENSP00000407546"}\NormalTok{,        }\StringTok{"CHRNA7"}\NormalTok{,        }\StringTok{"CHRNA7"}\NormalTok{,}\CommentTok{#}
\StringTok{"107987478"}\NormalTok{,                }\OtherTok{NA}\NormalTok{,                     }\OtherTok{NA}\NormalTok{,              }\OtherTok{NA}\NormalTok{,  }\StringTok{"LOC107987478"}\NormalTok{,}\CommentTok{#}
\StringTok{"107987479"}\NormalTok{,                }\OtherTok{NA}\NormalTok{,                     }\OtherTok{NA}\NormalTok{,              }\OtherTok{NA}\NormalTok{,  }\StringTok{"LOC107987479"}\NormalTok{,}\CommentTok{#}
     \StringTok{"1564"}\NormalTok{, }\StringTok{"ENSG00000205702"}\NormalTok{,                     }\OtherTok{NA}\NormalTok{,              }\OtherTok{NA}\NormalTok{,        }\StringTok{"CYP2D7"}\NormalTok{,}\CommentTok{#}
      \StringTok{"801"}\NormalTok{, }\StringTok{"ENSG00000198668"}\NormalTok{, }\StringTok{"9606.ENSP00000349467"}\NormalTok{,         }\StringTok{"CALM1"}\NormalTok{,         }\StringTok{"CALM1"}\NormalTok{,}\CommentTok{#}
      \StringTok{"805"}\NormalTok{, }\StringTok{"ENSG00000143933"}\NormalTok{, }\StringTok{"9606.ENSP00000272298"}\NormalTok{,         }\StringTok{"CALM2"}\NormalTok{,         }\StringTok{"CALM2"}\NormalTok{,}\CommentTok{#}
      \StringTok{"808"}\NormalTok{, }\StringTok{"ENSG00000160014"}\NormalTok{, }\StringTok{"9606.ENSP00000291295"}\NormalTok{,         }\StringTok{"CALM3"}\NormalTok{,         }\StringTok{"CALM3"} \CommentTok{#}
\NormalTok{)}\CommentTok{#########################################################################################}

\CommentTok{# Removing incomplete cases and adding updated ones}
\NormalTok{gene_ids }\OperatorTok{%<>%}\StringTok{ }\NormalTok{na.omit }\OperatorTok{%>%}\StringTok{ }\KeywordTok{bind_rows}\NormalTok{(complete_info)}

\CommentTok{# Removing taxid prefix from STRING IDs}
\NormalTok{gene_ids[[}\StringTok{"string_id"}\NormalTok{]] }\OperatorTok{%<>%}\StringTok{ }\KeywordTok{str_split_n}\NormalTok{(}\StringTok{"}\CharTok{\textbackslash{}\textbackslash{}}\StringTok{."}\NormalTok{, }\DecValTok{2}\NormalTok{)}

\CommentTok{# Exporting for package use}
\NormalTok{usethis}\OperatorTok{::}\KeywordTok{use_data}\NormalTok{(gene_ids, }\DataTypeTok{overwrite =} \OtherTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

