\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(here)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## here() starts at /home/danilo/R/neuro
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(purrr)}
\KeywordTok{library}\NormalTok{(tibble)}

\KeywordTok{library}\NormalTok{(knitr)}
\KeywordTok{library}\NormalTok{(kableExtra)}

\CommentTok{# download_if_missing <- function(filename, url) \{}
\CommentTok{#   if (!file.exists(here("data-raw", "download", filename))) \{}
\CommentTok{#     download.file(url, filename)}
\CommentTok{#   \}}
\CommentTok{# \}}

\CommentTok{# files <- tribble(}
\CommentTok{#   ~filename,           ~url,}
\CommentTok{#   "species.v11.0.txt", "https://stringdb-static.org/download/species.v11.0.txt",}
\CommentTok{#   # "download/taxonomy", "https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/new_taxdump/new_taxdump.tar.gz",}
\CommentTok{#   # "x",                 "y",}
\CommentTok{#   # "x",                 "y",}
\CommentTok{# )}

\NormalTok{files <-}\StringTok{ }\KeywordTok{tribble}\NormalTok{(}
  \OperatorTok{~}\NormalTok{url,}
  \OperatorTok{~}\NormalTok{filename,}
  
  \StringTok{"https://stringdb-static.org/download/species.v11.0.txt"}\NormalTok{,}
  \StringTok{"species.v11.0.txt"}\NormalTok{,}
  
  \StringTok{"https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/new_taxdump/new_taxdump.tar.gz"}\NormalTok{,}
  \StringTok{"new_taxdump.tar.gz"}\NormalTok{,}
  
  \StringTok{"http://rest.kegg.jp/link/pathway/hsa"}\NormalTok{,}
  \StringTok{"link_pathway_entrez.tsv"}\NormalTok{,}
  
  \StringTok{"https://string-db.org/mapping_files/entrez/human.entrez_2_string.2018.tsv.gz"}\NormalTok{,}
  \StringTok{"human.entrez_2_string.2018.tsv.gz"}\NormalTok{,}
  
  \StringTok{"https://string-db.org/mapping_files/STRING_display_names/human.name_2_string.tsv.gz"}\NormalTok{,}
  \StringTok{"human.name_2_string.tsv.gz"}\NormalTok{,}
  
  \StringTok{"https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz"}\NormalTok{,}
  \StringTok{"Homo_sapiens.gene_info.gz"}\NormalTok{,}
  
  \StringTok{"http://rest.genome.jp/link/ensembl/hsa"}\NormalTok{,}
  \StringTok{"link_ensembl_entrez.tsv"}\NormalTok{,}
  
  \CommentTok{# "https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz",}
  \CommentTok{# "Homo_sapiens.gene_info.gz",}
\NormalTok{)}

\NormalTok{files }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{kable}\NormalTok{(}\DataTypeTok{format =} \StringTok{"html"}\NormalTok{) }\OperatorTok{%>%}
\StringTok{  }\KeywordTok{kable_styling}\NormalTok{(}\DataTypeTok{bootstrap_options =} \StringTok{"striped"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

url

filename

\url{https://stringdb-static.org/download/species.v11.0.txt}

species.v11.0.txt

\url{https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/new_taxdump/new_taxdump.tar.gz}

new\_taxdump.tar.gz

\url{http://rest.kegg.jp/link/pathway/hsa}

link\_pathway\_entrez.tsv

\url{https://string-db.org/mapping_files/entrez/human.entrez_2_string.2018.tsv.gz}

human.entrez\_2\_string.2018.tsv.gz

\url{https://string-db.org/mapping_files/STRING_display_names/human.name_2_string.tsv.gz}

human.name\_2\_string.tsv.gz

\url{https://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz}

Homo\_sapiens.gene\_info.gz

\url{http://rest.genome.jp/link/ensembl/hsa}

link\_ensembl\_entrez.tsv

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# pwalk(files, download_if_missing)}
\KeywordTok{plot}\NormalTok{(}\DecValTok{1}\NormalTok{,}\DecValTok{1}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{figs/unnamed-chunk-2-1.pdf}

\hypertarget{special-cases}{%
\section{Special cases}\label{special-cases}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# if (!file.exists("download/taxonomy")) \{}
\CommentTok{#   tmp <- tempfile()}
\CommentTok{#   download.file("https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/new_taxdump/new_taxdump.tar.gz", tmp)}
\CommentTok{#   untar(tmp, exdir = "download/taxonomy")}
\CommentTok{#   unlink(tmp)}
\CommentTok{# \}}
\end{Highlighting}
\end{Shaded}

