---
title: "Supplementary Information"
subtitle: "Ionotropic receptors as the driving force behind human synapse establishment"
author: 
  - Lucas H. Viscardi
  - Danilo O. Imparato
  - Maria Cátira Bortolini
  - Rodrigo J. S. Dalmolin
  
keywords:
  - Evolution
  - Nervous system
   
header-includes:
  # kableExtra
  - \usepackage{booktabs}
  - \usepackage{longtable}
  - \usepackage{array}
  - \usepackage{multirow}
  - \usepackage{wrapfig}
  - \usepackage{float}
  - \usepackage{colortbl}
  - \usepackage{pdflscape}
  - \usepackage{tabu}
  - \usepackage{threeparttable}
  - \usepackage{threeparttablex}
  - \usepackage[normalem]{ulem}
  - \usepackage{makecell}
  - \usepackage{xcolor}
  - \usepackage{subfig}
  # Table caption left alignment
  - \usepackage{caption}
  - \captionsetup{justification=raggedright,singlelinecheck=false}
  - \captionsetup{margin={2pt,0pt}}
  # Hold figure position
  - \floatplacement{figure}{H}
  # Line numbers
  # - \usepackage{lineno}
  # - \linenumbers
  # Fira code
  - \setmonofont[Path=style/,Contextuals={Alternate},AutoFakeSlant]{FiraCode-Light.ttf}
  - \makeatletter
  - \def\verbatim@nolig@list{}
  - \makeatother
  # Small font size for output chunks
  - \let\oldverbatim\verbatim
  - \let\endoldverbatim\endverbatim
  - \renewenvironment{verbatim}{\scriptsize\oldverbatim}{\endoldverbatim}
  # Small font size for code chunks
  - \let\oldShaded\Shaded
  - \let\endoldShaded\endShaded
  # - \renewenvironment{Shaded}{\scriptsize\oldShaded}{\endoldShaded}
  # Rounded code blocks
  - \usepackage{tcolorbox}
  - \tcbuselibrary{breakable, skins}
  - \definecolor{chaos-bg}{HTML}{161616}
  - \newtcolorbox{blackbox}{colback=chaos-bg,frame hidden,boxsep=0pt,top=10pt,bottom=10pt,arc=6pt,enhanced jigsaw,breakable}
  - \renewenvironment{Shaded}{\scriptsize\begin{blackbox}}{\end{blackbox}}
  # emoji
  - \newfontfamily{\emojifont}[Path=style/]{seguiemj.ttf}
  - \DeclareTextFontCommand{\e}{\emojifont}
  # Supplementary Figure label
  - \renewcommand{\figurename}{Supplementary Figure}
  

geometry: margin=0.8in

graphics: yes

output:
  pdf_document:
    toc: yes
    toc_depth: 3
    fig_caption: true
    latex_engine: xelatex
    template: style/default.tex
    pandoc_args: --highlight-style=style/chaos_modified.theme
    keep_tex: true
---
```{r setup, include=FALSE}
gc()
knitr::knit_meta(clean = TRUE)
knitr::opts_chunk$set(fig.pos = 'H')
# knitr::opts_chunk$set(class.source = "numberLines")
```

<!--- Project structure -->
Project structure
=====================
This project is organized as an R package research compendium. Research compendia aim to provide a *"standard and easily recognizable way for organizing the digital materials of a research project to enable other researchers to inspect, reproduce, and extend the research"* [(Marwick, Boettiger &  Mullen, 2018)](https://doi.org/10.1080/00031305.2017.1375986). Project structure overview:  

`└──`\e{📦} `neurotransmissionevolution/`  
      `├──`\e{📂} `analysis/`  
      `│`    `├──`\e{📁} `plots/`  
      `│`    `├──`\e{📓} `geneplast.Rmd`  
      `│`    `├──`\e{📄} `geneplast_roots.tsv`  
      `│`    `└──` `...`  
      `├──`\e{📂} `data/`  
      `│`    `├──`\e{💾} `cogs.rda`  
      `│`    `├──`\e{📄} `gene_annotation.tsv`  
      `│`    `└──` `...`  
      `├──`\e{📂} data-`raw/`  
      `│`    `├──`\e{📓} `tree_ncbi.Rmd`  
      `│`    `├──`\e{📄} `tree_ncbi.nwk`  
      `│`    `└──` `...`  
      `├──`\e{📁} `inst/`  
      `├──`\e{📁} `man/`  
      `├──`\e{📂} `manuscripts/`  
      `│`    `├──`\e{📓} `supplement.Rmd`  
      `│`    `├──`\e{📰} `supplement.pdf`  
      `│`    `└──` `...`  
      `├──`\e{📁} `R/`  
      `│`    `├──`\e{📄} `utils.R`  
      `│`    `└──` `...`  
      `├──`\e{📄} `DESCRIPTION`  
      `├──`\e{📄} `NAMESPACE`  
      `├──`\e{®} `neurotransmissionevolution.Rproj`  
      `└──`\e{📄} `README.md`  

The main manuscript analysis is inside `analysis/` and uses preprocessed data from `data/`. Preprocessing steps are included in `data-raw/`. Everything is documented in RMarkdown. Some miscellaneous utility functions reside in `R/`. Additionally, both preprocessed data and utility functions are documented by roxygen2 and can be examined by usual help commands.
To explore and tweak this project, users can download the repository's latest release on github ("daniloimparato/neurotransmissionevolution@v1.0.0") and run `devtools::install()`.  

<!-- Preprocessing -->
Preprocessing
=================
This section mainly refers to data preprocessing steps done before the actual analysis with the intent of making it either simpler or faster.

<!--- Eukaryota species tree -->
Eukaryota species tree
--------------------------
We opted to use the TimeTree database to obtain an standardized Eukaryota species tree, although some target species were absent from it. Absent species were filled in based on NCBI Taxonomy data.

<!-- NCBI Taxonomy tree -->
### NCBI Taxonomy tree
The following topics describe how we retrieved a target species tree from NCBI Taxonomy data.
\input{data-raw.tree_ncbi.tex}

<!-- Hybrid tree -->
### Hybrid tree
Once we have both the NCBI Eukaryota tree and the list of duplicated genera, we can start assembling the complete hybrid tree.
\input{data-raw.tree_hybrid.tex}

<!--- Kegg annotation -->
Gene selection and annotation
---------------------------------
The anchoring point of our analysis is gene identifiers. This section describes the process of structuring an identifier dictionary as well as other useful annotation, like KEGG pathways. The goal is to have a table to which all kinds of additional data can be left joined.
\input{data-raw.kegg.tex}

<!--- Neuroexclusivity -->
Neuroexclusivity
--------------------
Neuroexclusivity data consists of gene expression collected from Gene Expression Atlas and the KEGG pathways themselves. This data is cleaned and subsetted prior to analysis.
\input{data-raw.neuroexclusivity.tex}

<!--- COG data -->
Orthology data
--------------------
This section refers to orthology data exported for geneplast use. Essentially, we subset the global STRING mapping between proteins and orthologous groups into a smaller dataset containing only information about orthogroups related to eukaryotic species.
\input{data-raw.cogs.tex}

<!--- Network -->
Network
-----------
We use our gene dictionary to retrieve the corresponding human protein-protein interaction network. This is done through the STRING API. In the end, we recompute combined interaction scores.
\input{data-raw.string_edgelist.tex}


<!-- Analysis -->
Analysis
============
This section focuses on analyses and results reported in the main manuscript.

Root inference
---------------
To estimate the evolutionary root of a given gene (i.e. the ancestor from which its genetic archetype was vertically inherited), we use orthologous group annotation from the STRING database. The presence and absence of an orthologous group in a species tree is used to determine its most likely evolutionary root.
\input{analysis.geneplast.tex}

<!--- Neuroexclusivity -->
Neuroexclusivity
-----------------
We characterize the relevance of a gene to the nervous system by inspecting what proportion of its activity is related to nervous processes. To do this, we rely on tissue RNA-Seq data, as well as KEGG's pathways themselves.
\input{analysis.neuroexclusivity.tex}

<!--- Network -->
Network
-----------
This section describes the steps for reproducing most of the manuscript figures. A lot of it is concerned with plotting particularities and visual styling.
\input{analysis.network.tex}

<!--- Abundance -->
Abundance
-----------
Abundance is the average number of proteins in neurotransmission orthogroups present in a species. This abundance is decomposed by orthogroup function. The orthogroup function is considered to be the function of its humans proteins in neurotransmission. 
\input{analysis.abundance.tex}


```{r end}
#
```
