---
title: "Supplementary Material"
subtitle: "Ionotropic receptors as the driving force behind human synapse establishment"
author: 
  - Lucas H. Viscardi
  - Danilo O. Imparato
  - Maria Cátira Bortolini
  - Rodrigo J. S. Dalmolin
  
keywords:
  - Evolution
  - Nervous system
   
header-includes:
  # kableExtra
  - \usepackage{booktabs}
  - \usepackage{longtable}
  - \usepackage{array}
  - \usepackage{multirow}
  - \usepackage{wrapfig}
  - \usepackage{float}
  - \usepackage{colortbl}
  - \usepackage{pdflscape}
  - \usepackage{tabu}
  - \usepackage{threeparttable}
  - \usepackage{threeparttablex}
  - \usepackage[normalem]{ulem}
  - \usepackage{makecell}
  - \usepackage{xcolor}
  - \usepackage{subfig}
  # Table caption left alignment
  - \usepackage{caption}
  - \captionsetup{justification=raggedright,singlelinecheck=false}
  - \captionsetup{margin={2pt,0pt}}
  # Hold figure position
  - \floatplacement{figure}{H}
  # Line numbers
  - \usepackage{lineno}
  - \linenumbers
  # Fira code
  - \setmonofont[Path=style/,Contextuals={Alternate},AutoFakeSlant]{FiraCode-Light.ttf}
  - \makeatletter
  - \def\verbatim@nolig@list{}
  - \makeatother
  # Small font size for output chunks
  - \let\oldverbatim\verbatim
  - \let\endoldverbatim\endverbatim
  - \renewenvironment{verbatim}{\scriptsize\oldverbatim}{\endoldverbatim}
  # Small font size for code chunks
  - \let\oldShaded\Shaded
  - \let\endoldShaded\endShaded
  # - \renewenvironment{Shaded}{\scriptsize\oldShaded}{\endoldShaded}
  # Rounded code blocks
  - \usepackage{tcolorbox}
  - \tcbuselibrary{breakable, skins}
  - \definecolor{chaos-bg}{HTML}{161616}
  - \newtcolorbox{blackbox}{colback=chaos-bg,frame hidden,boxsep=0pt,top=10pt,bottom=10pt,arc=6pt,enhanced jigsaw,breakable}
  - \renewenvironment{Shaded}{\scriptsize\begin{blackbox}}{\end{blackbox}}
  # emoji
  - \newfontfamily{\emojifont}[Path=style/]{seguiemj.ttf}
  - \DeclareTextFontCommand{\e}{\emojifont}
  

geometry: margin=1in

graphics: yes

output:
  pdf_document:
    toc: yes
    toc_depth: 3
    fig_caption: true
    latex_engine: xelatex
    template: style/default.tex
    pandoc_args: --highlight-style=style/chaos_modified.theme
    keep_tex: true
---
```{r setup, include=FALSE}
gc()
knitr::knit_meta(clean = TRUE)
knitr::opts_chunk$set(fig.pos = 'H')
knitr::opts_chunk$set(class.source = "numberLines")
```

<!--- Project structure -->
Project structure
=====================
This project is organized as an R package research compendium. Research compendia aim to provide a *"standard and easily recognizable way for organizing the digital materials of a research project to enable other researchers to inspect, reproduce, and extend the research"* [(Marwick, Boettiger &  Mullen, 2018)](https://doi.org/10.1080/00031305.2017.1375986). Project structure overview:  

`└──`\e{📦} `neurotransmissionevolution/`  
      `├──`\e{📂} `analysis/`  
      `│`    `├──`\e{📁} `plots/`  
      `│`    `├──`\e{📓} `geneplast.Rmd`  
      `│`    `├──`\e{📄} `geneplast_roots.tsv`  
      `│`    `└──` `...`  
      `├──`\e{📂} `data/`  
      `│`    `├──`\e{💾} `cogs.rda`  
      `│`    `├──`\e{📄} `gene_annotation.tsv`  
      `│`    `└──` `...`  
      `├──`\e{📂} data-`raw/`  
      `│`    `├──`\e{📓} `tree_ncbi.Rmd`  
      `│`    `├──`\e{📄} `tree_ncbi.nwk`  
      `│`    `└──` `...`  
      `├──`\e{📁} `inst/`  
      `├──`\e{📁} `man/`  
      `├──`\e{📂} `manuscripts/`  
      `│`    `├──`\e{📓} `supplement.Rmd`  
      `│`    `├──`\e{📰} `supplement.pdf`  
      `│`    `└──` `...`  
      `├──`\e{📁} `R/`  
      `│`    `├──`\e{📄} `utils.R`  
      `│`    `└──` `...`  
      `├──`\e{📄} `DESCRIPTION`  
      `├──`\e{📄} `NAMESPACE`  
      `├──`\e{®} `neurotransmissionevolution.Rproj`  
      `└──`\e{📄} `README.md`

<!-- Preprocessing --The main manuscript analysis is inside `analysis/` and uses preprocessed data from `data/`. Preprocessing steps are included in `data-raw/`. Everything is documented in RMarkdown.  


>
Preprocessing
=================
This topic refers mainly to data wrangling done before the actual analysis with the intent of making it simpler.

<!--- Eukaryota species tree -->
Eukaryota species tree
--------------------------
We opted to use the TimeTree database in order to obtain an standardized Eukaryota species tree. However, some species were not present in it, so we devised a way to fill them in based on NCBI Taxonomy data. 

<!-- NCBI Taxonomy tree -->
### NCBI Taxonomy tree
First we preprocess NCBI Taxonomy data to leave only STRING eukaryotes, thus making the task easier.
\input{data-raw.tree_ncbi.tex}
<!-- \input{download_files.tex} -->

<!-- Duplicated Genera -->
### Duplicated Genera
Some species from different kingdoms may share the same genus name. These genera must be noted down because one of the ways we fill in missing species is by looking at genera names.
\input{data-raw.duplicated_genera.tex}

<!-- Hybrid tree -->
### Hybrid tree
Once we have both the NCBI eukaryotes tree and the list of duplicated genera, we can start assembling the complete hybrid tree.
\input{data-raw.tree_hybrid.tex}

<!--- Kegg annotation -->
Gene selection and annotation
---------------------------------
The anchoring point for this study is basic annotation about genes and the pathways in which they participate. This section describes the process of structuring such data. In the end we will have a table to which all kinds of additional data will be left joined into.
\input{data-raw.kegg.tex}

<!--- Neuroexclusivity -->
Neuroexclusivity
--------------------
Neuroexclusivity data consists of gene expression collected from Gexe Expression Atlas and the KEGG pathways themselves.
\input{data-raw.neuroexclusivity.tex}

<!--- COG data -->
Orthology data
--------------------
This section refers to orthology data exported for geneplast use. Essentialy, we subset the global STRING mapping between proteins and orthologous groups into a smaller dataset containing only information about the orthogroups related to our selected genes.
\input{data-raw.cogs.tex}

<!--- Network -->
Network
-----------
In this section we search the STRING API for our proteins of interest and recompute the combined interaction score.
\input{data-raw.string_edgelist.tex}


<!-- Analysis -->
Analysis
============
Analysis

Root inference
---------------
To estimate the evolutionary root of a given gene, i.e. the ancestor from which its genetic archetype (orthologous group) is vertically inherited, we use orthologous group annotation from the STRING database. The presence and absence of orthologous groups in the species of a cladogram are used to determine its most likely ancestor.
\input{analysis.geneplast.tex}

<!--- Neuroexclusivity -->
Neuroexclusivity
-----------------
We characterize genes' relevance to the nervous system by inspecting what proportion of its activity is related to nervous processes. We relied on tissue RNA-Seq data, as well as KEGG's pathways themselves.
\input{analysis.neuroexclusivity.tex}

<!--- Network -->
Network
-----------
In this section we search the STRING API for our proteins of interest and recompute their combined interaction score.
\input{analysis.network.tex}

<!--- Abundance -->
Abundance
-----------
Abundance is the average number of proteins in neurotransmission orthogroups present in a species. This abundance is decomposed by orthogroup function. The orthogroup function is considered to be the function of its humans proteins in neurotransmission, as described in previous sections. In other words: "what would be the average number of "X"-related proteins in neurotransmission COGs in a species?"
\input{analysis.abundance.tex}


```{r end}
#
```
