---
# layout: 12pt,review
linenumbers: true
title: "Ionotropic receptors as the driving force behind human synapse establishment"
subtitle: "Supplementary Material"
author: 
  - Lucas H. Viscardi
  - Danilo O. Imparato
  - Maria CÃ¡tira Bortolini
  - Rodrigo J. S. Dalmolin
abstract: |
  Model uncertainty and limited data are fundamental challenges to
  robust management of human intervention in a natural system.
  These challenges are acutely highlighted by concerns that
  many ecological systems may contain tipping points, such as
  Allee population sizes.  Before a collapse, we do not know
  where the tipping points lie, if they exist at all.  Hence,
  we know neither a complete model of the system dynamics nor
  do we have access to data in some large region of state-space
  where such a tipping point might exist.
  
keywords:
  - Evolution
  - Nervous system
   
header-includes:
  - \usepackage{booktabs}
  - \usepackage{longtable}
  - \usepackage{array}
  - \usepackage{multirow}
  - \usepackage{wrapfig}
  - \usepackage{float}
  - \usepackage{colortbl}
  - \usepackage{pdflscape}
  - \usepackage{tabu}
  - \usepackage{threeparttable}
  - \usepackage{threeparttablex}
  - \usepackage[normalem]{ulem}
  - \usepackage{makecell}
  - \usepackage{xcolor}
  # small font size for code chunks
  - \let\oldShaded\Shaded
  - \let\endoldShaded\endShaded
  - \renewenvironment{Shaded}{\scriptsize\oldShaded}{\endoldShaded}
  # small font size for output chunks
  - \let\oldverbatim\verbatim
  - \let\endoldverbatim\endverbatim
  - \renewenvironment{verbatim}{\scriptsize\oldverbatim}{\endoldverbatim}
  # table caption left alignment
  - \usepackage{caption}
  - \captionsetup{justification=raggedright,singlelinecheck=false}
  - \captionsetup{margin={2pt,0pt}}
  # hold figure position
  - \floatplacement{figure}{H}

output:
  pdf_document:
    toc: yes
    toc_depth: 3
    fig_caption: true
    highlight: zenburn
    # pandoc_args: "--highlight=breezedark"
    # template: components/elsarticle.latex
    keep_tex: true
---
```{r setup, include=FALSE}
#a
```

<!--- Project structure -->
**Project structure**
=====================
This is the title page

**Preprocessing**
=================
This topic refers mainly to data wrangling done before the actual analysis with the intent of making it simpler.

<!--- Eukaryota species tree -->
**Eukaryota species tree**
--------------------------
We opted to use the TimeTree database in order to obtain an standardized Eukaryota species tree. However, some species were not present in it, so we devised a way to fill them in based on NCBI Taxonomy data. 

### **NCBI Taxonomy tree**
First we preprocess NCBI Taxonomy data to leave only STRING eukaryotes, thus making the task easier.
\input{tree_ncbi.tex}
<!-- \input{download_files.tex} -->

### **Duplicated Genera**
Some species from different kingdoms may share the same genus name. These genera must be noted down because one of the ways we fill in missing species is by looking at genera names.
\input{duplicated_genera.tex}

### **Hybrid tree**
Once we have both the NCBI eukaryotes tree and the list of duplicated genera, we can start assembling the complete hybrid tree.
\input{tree_hybrid.tex}

<!--- Kegg annotation -->
**Gene selection and annotation**
---------------------------------
The anchoring point for this study is basic annotation about genes and the pathways in which they participate. This section describes the process of structuring such data. In the end we will have a table to which all kinds of additional data will be left joined into.
\input{kegg.tex}

<!--- Neuroexclusivity -->
**Neuroexclusivity**
--------------------
Explanation

### **Expression**

### **Pathways**

<!--- COG data -->
**COG data**
------------

<!--- Network -->
**Network**
-----------

**Analysis**
============
Analysis

<!-- \input{download_files.tex} -->

```{r end}
#leave this chunk
```
