% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Ionotropic receptors as the driving force behind human synapse establishment},
  pdfauthor={Lucas H. Viscardi; Danilo O. Imparato; Maria Cátira Bortolini; Rodrigo J. S. Dalmolin},
  pdfkeywords={Evolution, Nervous system},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{22,22,22}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{1.00,0.81,0.69}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.50,0.62,0.50}{\textbf{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.82,0.82,0.82}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.86,0.64,0.64}{#1}}
\newcommand{\BuiltInTok}[1]{\textcolor[rgb]{0.82,0.82,0.82}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.86,0.64,0.64}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.33,0.33,0.33}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.50,0.62,0.50}{\textbf{#1}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.86,0.64,0.64}{\textbf{#1}}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.94,0.87,0.69}{#1}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.87,0.87,0.75}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.86,0.86,0.80}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.50,0.62,0.50}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.76,0.75,0.62}{#1}}
\newcommand{\ExtensionTok}[1]{\textcolor[rgb]{0.82,0.82,0.82}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.75,0.75,0.82}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.94,0.94,0.56}{#1}}
\newcommand{\ImportTok}[1]{\textcolor[rgb]{0.82,0.82,0.82}{#1}}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.50,0.62,0.50}{\textbf{#1}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.94,0.87,0.69}{#1}}
\newcommand{\NormalTok}[1]{\textcolor[rgb]{0.82,0.82,0.82}{#1}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.94,0.94,0.82}{#1}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.94,0.94,0.56}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{1.00,0.81,0.69}{\textbf{#1}}}
\newcommand{\RegionMarkerTok}[1]{\textcolor[rgb]{0.82,0.82,0.82}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.86,0.64,0.64}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.80,0.58,0.58}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.80,0.58,0.58}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.82,0.82,0.82}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.80,0.58,0.58}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.50,0.62,0.50}{\textbf{#1}}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}
\usepackage{subfig}
\let\oldShaded\Shaded
\let\endoldShaded\endShaded
\renewenvironment{Shaded}{\scriptsize\oldShaded}{\endoldShaded}
\let\oldverbatim\verbatim
\let\endoldverbatim\endverbatim
\renewenvironment{verbatim}{\scriptsize\oldverbatim}{\endoldverbatim}
\usepackage{caption}
\captionsetup{justification=raggedright,singlelinecheck=false}
\captionsetup{margin={2pt,0pt}}
\floatplacement{figure}{H}
\usepackage{lineno}
\linenumbers
\setmonofont[Path=style/,Contextuals={Alternate},AutoFakeSlant]{FiraCode-Light.ttf}
\makeatletter
\def\verbatim@nolig@list{}
\makeatother
\usepackage{tcolorbox}
\tcbuselibrary{breakable, skins}
\definecolor{chaos-bg}{HTML}{161616}
\newtcolorbox{blackbox}{colback=chaos-bg,colframe=gray,boxsep=2pt,arc=4pt,enhanced jigsaw,breakable}
\renewenvironment{Shaded}{\scriptsize\begin{blackbox}}{\end{blackbox}}

\title{Ionotropic receptors as the driving force behind human synapse
establishment}
\usepackage{etoolbox}
\makeatletter
\providecommand{\subtitle}[1]{% add subtitle to \maketitle
  \apptocmd{\@title}{\par {\large #1 \par}}{}{}
}
\makeatother
\subtitle{Supplementary Material}
\author{Lucas H. Viscardi \and Danilo O. Imparato \and Maria Cátira Bortolini \and Rodrigo J. S. Dalmolin}
\date{}

\begin{document}
\maketitle
\begin{abstract}
The origin of nervous systems is a main theme in biology and its
mechanisms are largely underlied by synaptic neurotransmission. The
search for pan-neuronal genes has failed to explain synapses emergence
since synaptic elements are present in multiple aneural organisms. We
questioned how the interactions among these elements evolved and to what
extent does it relate to our understanding of the nervous systems
complexity. We infer the human neurotransmission gene network based on
genes present in GABAergic, glutamatergic, serotonergic, dopaminergic,
and cholinergic systems, and reconstruct the evolutionary scenario of
synapse emergence. Our results outline the distribution of
neurotransmitter systems and synaptic functions as taxa diverge. We find
that the ionotropic receptors emergence in the cnidarian last common
ancestor was the driving force behind the advent of the anatomical
synapse and the development of nervous systems. We suggest that after
Placozoa origin there was a positive selection under a more restricted
and specific activation of ionotropic receptors. This different
selective pressure counterparting the broader amino acid activation in
plants could be striking for the evolution of the nervous system.
\end{abstract}

{
\setcounter{tocdepth}{3}
\tableofcontents
}
\hypertarget{project-structure}{%
\section{Project structure}\label{project-structure}}

This is the title page

\hypertarget{preprocessing}{%
\section{Preprocessing}\label{preprocessing}}

This topic refers mainly to data wrangling done before the actual
analysis with the intent of making it simpler.

\hypertarget{eukaryota-species-tree}{%
\subsection{Eukaryota species tree}\label{eukaryota-species-tree}}

We opted to use the TimeTree database in order to obtain an standardized
Eukaryota species tree. However, some species were not present in it, so
we devised a way to fill them in based on NCBI Taxonomy data.

\hypertarget{ncbi-taxonomy-tree}{%
\subsubsection{NCBI Taxonomy tree}\label{ncbi-taxonomy-tree}}

First we preprocess NCBI Taxonomy data to leave only STRING eukaryotes,
thus making the task easier. \input{data-raw.tree_ncbi.tex}

\hypertarget{duplicated-genera}{%
\subsubsection{Duplicated Genera}\label{duplicated-genera}}

Some species from different kingdoms may share the same genus name.
These genera must be noted down because one of the ways we fill in
missing species is by looking at genera names.
\input{data-raw.duplicated_genera.tex}

\hypertarget{hybrid-tree}{%
\subsubsection{Hybrid tree}\label{hybrid-tree}}

Once we have both the NCBI eukaryotes tree and the list of duplicated
genera, we can start assembling the complete hybrid tree.
\input{data-raw.tree_hybrid.tex}

\hypertarget{gene-selection-and-annotation}{%
\subsection{Gene selection and
annotation}\label{gene-selection-and-annotation}}

The anchoring point for this study is basic annotation about genes and
the pathways in which they participate. This section describes the
process of structuring such data. In the end we will have a table to
which all kinds of additional data will be left joined into.
\input{data-raw.kegg.tex}

\hypertarget{neuroexclusivity}{%
\subsection{Neuroexclusivity}\label{neuroexclusivity}}

Neuroexclusivity data consists of gene expression collected from Gexe
Expression Atlas and the KEGG pathways themselves.
\input{data-raw.neuroexclusivity.tex}

\hypertarget{orthology-data}{%
\subsection{Orthology data}\label{orthology-data}}

This section refers to orthology data exported for geneplast use.
Essentialy, we subset the global STRING mapping between proteins and
orthologous groups into a smaller dataset containing only information
about the orthogroups related to our selected genes.
\input{data-raw.cogs.tex}

\hypertarget{network}{%
\subsection{Network}\label{network}}

In this section we search the STRING API for our proteins of interest
and recompute the combined interaction score.
\input{data-raw.string_edgelist.tex}

\hypertarget{analysis}{%
\section{Analysis}\label{analysis}}

Analysis

\hypertarget{root-inference}{%
\subsection{Root inference}\label{root-inference}}

To estimate the evolutionary root of a given gene, i.e.~the ancestor
from which its genetic archetype (orthologous group) is vertically
inherited, we use orthologous group annotation from the STRING database.
The presence and absence of orthologous groups in the species of a
cladogram are used to determine its most likely ancestor.
\input{analysis.geneplast.tex}

\hypertarget{neuroexclusivity-1}{%
\subsection{Neuroexclusivity}\label{neuroexclusivity-1}}

We characterize genes' relevance to the nervous system by inspecting
what proportion of its activity is related to nervous processes. We
relied on tissue RNA-Seq data, as well as KEGG's pathways themselves.
\input{analysis.neuroexclusivity.tex}

\hypertarget{network-1}{%
\subsection{Network}\label{network-1}}

In this section we search the STRING API for our proteins of interest
and recompute their combined interaction score.
\input{analysis.network.tex}

\hypertarget{abundance}{%
\subsection{Abundance}\label{abundance}}

Abundance is the average number of proteins in neurotransmission
orthogroups present in a species. This abundance is decomposed by
orthogroup function. The orthogroup function is considered to be the
function of its humans proteins in neurotransmission, as described in
previous sections. In other words: ``what would be the average number
of''X``-related proteins in neurotransmission COGs in a species?''
\input{analysis.abundance.tex}

\begin{Shaded}
\begin{Highlighting}[numbers=left,,]
\CommentTok{#}
\end{Highlighting}
\end{Shaded}

\end{document}
