---
title: "Geneplast"
knit: neurotransmissionevolution::custom_knit
---
```{r include=FALSE}
knitr::opts_chunk$set(
   warning     = FALSE
  ,message     = FALSE
  ,cache       = 1
  ,autodep     = T
)
```

Loading initial resources:
```{r}
library(tidyverse)
library(magrittr)
library(geneplast)
library(ape)
library(XML)
library(rentrez)
library(neurotransmissionevolution)

data(
  cogs,
  gene_cogs,
  string_eukaryotes,
  package = "neurotransmissionevolution"
)

phyloTree <- read.tree("../data/hybrid_tree_modified.nwk") %>% rotatePhyloTree("9606")
```

We perform some minor data formatting before feeding it to geneplast
```{r}
# Formating cogdata column names for geneplast
cogs %<>% select(protein_id = string_id, ssp_id = taxid, cog_id)

# Adding species names to taxid tree
phyloTree %<>% list_modify(
  tip.alias = string_eukaryotes %$% string_name[match(phyloTree[["tip.label"]], taxid)]
)
```

### Geneplast
Geneplast's `groot.preprocess` function structures an `ogr` object on which `groot` will perform the rooting. We then retrieve the numeric root (`groot.get("results")`) for the `cogs_of_interest`, that is, orthologous groups pertaining to neurotransmission genes.
```{r results="hide"}
cogs_of_interest <- gene_cogs %>% pull(cog_id) %>% unique

ogr <- groot.preprocess(
  cogdata   = cogs,
  phyloTree = phyloTree,
  spid      = "9606",
  cogids    = cogs_of_interest
)

roots <- groot(ogr, nPermutations = 1) %>%
  groot.get("results") %>%
  rownames_to_column("cog_id") %>%
  select(cog_id, root = Root) %>%
  write_tsv("geneplast_roots.tsv")
```

### Clade names
Each root branches to a clade that diverged from humans some time in the past. It is nice to have these clades taxonomically named to ease our interpretation. Unlike NCBI Taxonomy, TimeTree's internal nodes are not named. Therefore, we query the NCBI Taxonomy API to try to find most clade names automatically. It is important to note that we are using a hybrid tree primarily built from TimeTree data. This means NCBI Taxonomy naming will not perfectly match clades in our tree. For instance, root #36 branches to a clade containing 38 species from the SAR supergroup, but also 1 species from the Haptista rank, namely _Emiliania huxleyi_. The Haptista group is a sister clade to SAR, so it might be the case that _Emiliania huxleyi_ is actually correctly placed together with SAR species by TimeTree, given their evolutionary proximity. Resolving these naming conflicts is not trivial and falls out of our scope.
```{r}
# Querying NCBI Taxonomy with our taxids
lineages <- entrez_fetch(
  db      = "taxonomy",
  id      = string_eukaryotes[["new_taxid"]],
  rettype = "xml",
  retmode = "xml",
  parsed  = TRUE
)

# Parsing the XML result and retrieving lineage data
string_eukaryotes %<>% mutate(
  root        = ogr@tree$tip.group[taxid],
  lineage_txt = xpathSApply(lineages, "//Lineage", XML::xmlValue)
)

# Writing all lineage data to manually check for edge cases
string_eukaryotes %>%
  select(root, lineage_txt) %>%
  arrange(root, lineage_txt) %>%
  write_tsv("temp/species_lineage.txt")

# The following chain of dplyr verbs
# is responsible for figuring out the best clade names
root_names <- string_eukaryotes %>%
  
  # Long format lineages
  mutate(
     lineage_split = strsplit(lineage_txt, "; ")
  ) %>%
  unnest_longer(
     col        = lineage_split
    ,values_to  = "clade_name"
    ,indices_to = "clade_depth"
  ) %>%
  
  # Counting and dropping the last group
  group_by(root, clade_depth, clade_name) %>%
  tally(sort = TRUE) %>%
  
  # Collapsing lineages by clade depths
  summarise(
     diverging_rank = n_distinct(clade_name) > 1
    ,clade_name     = ifelse(diverging_rank, paste0(clade_name, " (", n,")", collapse = "; "), clade_name)
  ) %>%
  
  # Removing diverging ranks after the first one
  filter(cumsum(diverging_rank) <= 1) %>%
  
  # Removing irrelevant basal ranks (eg Eukaryota)
  group_by(clade_depth) %>%
  arrange(root) %>%
  filter(!(duplicated(clade_name) | duplicated(clade_name, fromLast = TRUE)) | diverging_rank) %>%
  
  # Choosing name
  group_by(root) %>%
  summarise(clade_name = first(clade_name, order_by = clade_depth)) %>%
  write_tsv("temp/temp_geneplast_clade_names.tsv")
```

Some automatically named clades have to be resolved by hand. The following table shows clade names before and after manual checking:
```{r echo=FALSE}
# Loading manually resolved names, based on temp/temp_geneplast_clade_names.tsv
lca_names <- read_tsv("geneplast_clade_names.tsv")

root_names %>%
  rename("Automatic name" = clade_name) %>%
  inner_join(lca_names) %>%
  rename("Corrected name" = clade_name, "Root" = root) %>%
  knitr::kable(caption = "Clade names before and after manual checking.", booktabs = TRUE, linesep = "") %>%
  kableExtra::kable_styling(position = "left", latex_options = c("striped", "HOLD_position"))
```

### Phyletic patterns
Visualizing the presence/absence of COGs across species.
```{r phyletic_patterns, fig.show="hide", fig.height=31, fig.width=7}
lca_names %<>% rename("lca" = root)

lca_spp <- ogr@spbranches %>%
  rename("taxid" = ssp_id, "species" = ssp_name, "lca" = `9606`) %>%
  mutate(taxid_order = row_number())
  
# Saving for use in abundance computation
lca_spp %>% select(lca, taxid, taxid_order) %>% write_tsv("geneplast_clade_taxids.tsv")

cog_pam <- ogr@orthoct[,-1]

long_pam <- cog_pam %>%
  rownames_to_column("taxid") %>%
  pivot_longer(-taxid, names_to = "cog_id") %>%
  left_join(lca_spp) %>%
  left_join(lca_names) %>%
  left_join(roots) %>%
  mutate(
    cog_id       = fct_reorder(cog_id, root),
    species      = fct_reorder(species, desc(taxid_order)),
    clade_name   = fct_reorder(clade_name, lca),
    root         = as_factor(root),
    clade_stripe = as.numeric(as_factor(lca)) %% 2 == 0
  ) %>%
  # Stripe every other species
  group_by(cog_id) %>%
  mutate(spp_stripe = as.numeric(species) %% 2 == 0) %>%
  # Removing empty tiles
  filter(value == 1) %>%
  # Stripe every other cog
  group_by(taxid) %>%
  mutate(cog_stripe = as.numeric(cog_id) %% 2 == 0)

ggplot(long_pam, aes(x = cog_id, y = species)) +
  geom_tile(aes(fill = clade_stripe + 0.3 * xor(spp_stripe, cog_stripe))) +
  scale_fill_gradient(low = "#424242", high = "#212121") +
  facet_grid(clade_name ~ root, scales = "free", space = "free") +
  xlab("COGs") +
  ylab("Species") +
  theme(
     legend.position    = "none"
    ,strip.background   = element_blank()
    ,strip.text.x       = element_text(size = 4, angle = 0, vjust = 0)
    ,strip.text.y       = element_text(size = 5, angle = 0, hjust = 0)
    ,axis.text.x        = element_text(size = 3, angle = -45, vjust = 0, hjust = 0.125)
    ,axis.text.y        = element_text(size = 4)
    ,axis.title         = element_text(size = 8)
    ,axis.ticks         = element_line(size = 0.25)
    ,panel.grid.major.y = element_line(size = 0.25)
    ,panel.grid.major.x = element_blank()
    ,panel.spacing      = unit(0.25, "pt")
    ,plot.margin        = unit(c(0,0,0,0), "mm")
  )
```
\clearpage

\thispagestyle{empty}

\pdfpageheight=33in

\begin{figure}[p]

\caption{Presence of orthologous groups in species. The horizontal axis is grouped according to assigned roots. The vertical axis is grouped by clades. A checkerboard pattern is superimposed to aid visualization.}\label{fig:phyletic_patterns}

{\centering \includegraphics[height=31in, width=7in]{figs/analysis.geneplast.phyletic_patterns-1} }

\end{figure}

\clearpage

\pdfpageheight=11in

### Adhesion genes
Aditionally, we also infer the roots of neural adhesion genes.
```{r results="hide"}
data(
  adhesion_genes,
  gene_cogs_extra,
  package = "neurotransmissionevolution"
)

cogs_of_interest_extra <- gene_cogs_extra %>% pull(cog_id) %>% unique

ogr_extra <- groot.preprocess(
  cogdata   = cogs,
  phyloTree = phyloTree,
  spid      = "9606",
  cogids    = cogs_of_interest_extra
)

roots_extra <- groot(ogr_extra, nPermutations = 1) %>%
  groot.get("results") %>%
  rownames_to_column("cog_id") %>%
  left_join(gene_cogs_extra) %>%
  left_join(adhesion_genes) %>%
  left_join(lca_names, by = c("Root" = "lca"))
```
```{r results="asis", echo=FALSE}
roots_extra %>%
  arrange(Root, entrez_name) %>%
  mutate(Root = paste0("Human-", clade_name, " LCA (#", Root,")")) %>%
  select("String ID" = string_id, "COG" = cog_id, "Symbol" = entrez_name, Root) %>%
  knitr::kable(
    caption = "Adhesion genes in the neural system (KEGG Pathway hsa04514) and their roots."
    ,booktabs = TRUE
    ,longtable = T
    ,linesep = ""
  ) %>%
  kableExtra::kable_styling(position = "left", latex_options = c("striped"))
```
